server/src/main/java/org/opensearch/identity/noop/NoopAuthenticationManager.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.noop;
+
+import org.opensearch.identity.AuthenticationManager;
+import org.opensearch.identity.Subject;
+
+/**
+ * Implementation of authentication manager that does not enforce authentication
+ *
+ * This class and related classes in this package will not return nulls or fail permissions checks
+ */
+public class NoopAuthenticationManager implements AuthenticationManager {
+
+    @Override
+    public Subject getSubject() {
+        return new NoopSubject();
```

Comment:
Would it make sense to have a single instance of NoopSubject instead of recreating a new instance every time `getSubject` is called?

Reply:
It was less code this way, since the objects operate the same way this seems to be of little harm.

Reply:
i think we should return same object instead of new one